HACKING.rst

```diff
@@ -29,33 +29,38 @@ you may need to run::
   pip install importlib
 
 
-**Testing the HaaS**
-====================
+`Testing.md <docs/testing.md>`_ contains more information about testing HaaS.
 
+Configuring HaaS
+================
 
 Now the ``haas`` executable should be in your path.  First, create a
 configuration file ``haas.cfg``. There are two examples for you to work from,
-``haas.cfg.dev-example``, which is oriented towards developemnt, and
-``haas.cfg.example`` which is more production oriented. These files are well
-commented; read them carefully. In relation to testing, one option is worth
-calling out specifically: the ``dry_run`` option in the ``[devel]`` section,
-which allows most unit tests (which only touch the database) to be run on one's
-local machine, without needing to have a full HaaS setup available.
-
-Next initialize the database with the required tables, with ``haas
-init_db``.  Run the server with ``haas serve``.  Finally, see ``haas help``
-for the various API commands one can test.  Here is an example session,
-testing ``headnode_delete_hnic``::
+``examples/haas.cfg.dev-example``, which is oriented towards development, and
+``examples/haas.cfg.example`` which is more production oriented.
+
+HaaS can be configured to not perform state-changing operations on nodes,
+headnodes and networks, allowing developers to run and test parts of a haas
+server without requiring physical hardware. To supress actual node and headnode
+operations, set ``dry_run = True`` in the ``[devel]`` section. For supressing
+actual network switch operations, use either the ``null`` or ``null_vlan``
+network driver by setting either ``driver = null`` or ``driver = null_vlan``,
+respectively, in the ``[general]`` section.  When using ``null_vlan``, one must
+also set ``vlans = X-Y`` in the ``[vlan]`` section, such as ``vlans =
+100-200``.
+
+These config files are well commented; read them carefully. Next initialize the
+database with the required tables, with ``haas init_db``. Run the server with
+``haas serve`` and ``haas serve_networks`` in separate windows.  Finally, see
```

Comment:
Nitpick: separate terminals, not necessarily separate windows. Could be a screen session or something. Also, suggest "run ``haas help``" rather than see.